### PR TITLE
User pages and profile cards

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/components/User/Profiles/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+export type ProfileCardProps = {
+    user: TPublicUserRecord;
+    size?: ProfileSize;
+    status?: 'offline' | 'online' | 'ingame'; // TODO: proper localization for statuses.
+};
+
+export function ProfileCard(
+    _props: JSX.HTMLElementTags['a'] & ProfileCardProps
+) {
+    const [props, rest] = splitProps(_props, ['user', 'size', 'status']);
```
Comment (kearfy) [inline src/components/User/Profiles/Card.tsx]:
```suggestion
    const [props, rest] = splitProps(_props, ['user', 'size', 'status', 'class']);
```

File: src/components/User/Profiles/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    const user = () => props.user;
+    const profile = () => props.user.profile;
+    const name = () => `${profile().displayname || user().username}`;
+
+    return (
+        <a
+            href={`/profile/${user().username}`}
+            class={cardStyle({
+                size: props.size,
```
Comment (kearfy) [inline src/components/User/Profiles/Card.tsx:59]:
```suggestion
                size: props.size,
                class: props.class,
```

File: src/components/User/Profiles/Container.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import { Overview } from './Tabs/Overview';
+
+import styles from '~/styles/components/User/Profiles/Page.module.scss';
+
+export function ProfileContainer(
+    _props: JSX.HTMLElementTags['div'] & {
+        user: TPublicUserRecord;
+    }
+) {
+    const [props, rest] = splitProps(_props, ['user']);
```
Comment (kearfy) [inline src/components/User/Profiles/Container.tsx]:
```suggestion
    const [props, rest] = splitProps(_props, ['user', 'class']);
```

File: src/components/User/Profiles/Container.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    const user = () => props.user;
+
+    return (
+        <div class={styles.container} {...rest}>
```
Comment (kearfy) [inline src/components/User/Profiles/Container.tsx]:
```suggestion
        <div class={[styles.container, props.class].filter(a => a).join(' ')} {...rest}>
```

**Review by kearfy — CHANGES_REQUESTED** [review]
Pass those props as much as possible, keeps things flexible. Right now it would have been possible to pass class as a prop to the component, but it would not do anything.

**Review by ghost — APPROVED** [review]

**Review by kearfy — APPROVED** [review]

**Pull request merged**